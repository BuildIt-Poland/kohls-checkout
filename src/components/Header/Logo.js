import React from 'react';
import { Link as LinkBase } from 'react-router-dom';
import styled from 'styled-components';

import { Logo as LogoIcon } from '../Icons';

const Link = styled(LinkBase)`
  height: 100%;
  cursor: pointer;
  padding-top: 2rem;
`;

function Logo() {
  return (
    <Link to="/">
      <LogoIcon />
    </Link>
  );
}

export default Logo;
