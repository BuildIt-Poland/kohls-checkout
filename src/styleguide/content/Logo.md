This section contains the **The Store** logo but it can also have additional logos.

### Logo guidelines

- Place the logo on white background
- Never place the logo on images
- When positioning the logo beside elements, ensure there is enough clear space
- Don't add additional shadows or effects, modify colors or change orientation

It can be imported into the codebase like this:

`import { Logo } from '../Icons';`

```jsx noeditor
import styled from 'styled-components'

import { Logo } from '../../components/Icons';
import { colorLightGrey } from '../../styles/designTokens.js';

const LogoWrapper = styled.div`
  border: 1px solid ${colorLightGrey};
  padding: 30px;
  border-radius: 8px;
`;

<LogoWrapper>
  <Logo />
</LogoWrapper>
```
