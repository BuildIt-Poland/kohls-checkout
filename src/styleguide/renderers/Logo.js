import React, { Component } from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  background-color: #222;
`;

const Image = styled.img`
  width: 100%;
  height: 100%;
`;

class Logo extends Component {
  render() {
    return (
      <Wrapper>
        <Image src="styleguide-logo.svg" alt="The Store Styleguide by Buildit" />
      </Wrapper>
    );
  }
}

export default Logo;
