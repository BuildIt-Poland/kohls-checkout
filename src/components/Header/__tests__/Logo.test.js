import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Logo from '../Logo';

jest.mock('../../Icons', () => ({
  Logo: 'LogoIcon'
}));

describe('COMPONENT - Header Logo', () => {
  it('renders correctly', () => {
    const component = create(
      <MemoryRouter initialEntries={['/']} initialIndex={1}>
        <Logo />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
