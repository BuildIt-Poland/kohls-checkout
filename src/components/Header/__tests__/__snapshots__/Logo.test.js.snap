// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Header Logo renders correctly 1`] = `
.c0 {
  height: 100%;
  cursor: pointer;
  padding-top: 2rem;
}

<a
  className="c0"
  href="/"
  onClick={[Function]}
>
  <LogoIcon />
</a>
`;
